lib/ansible/plugins/action/unarchive.py

```diff
@@ -66,9 +66,7 @@ def run(self, tmp=None, task_vars=None):
             # do not run the command if the line contains creates=filename
             # and the filename already exists. This allows idempotence
             # of command executions.
-            result = self._execute_module(module_name='stat', module_args=dict(path=creates), task_vars=task_vars)
-            stat = result.get('stat', None)
-            if stat and stat.get('exists', False):
+            if self._remote_file_exists(creates):
```

Comment:
This code should not change it working or not, but it is more light weight and efficient.